sources/web/datalab/polymer/modules/gapi-manager/gapi-manager.ts

```diff
@@ -277,6 +277,165 @@ class ClientAuth implements GapiAuth {
   }
 }
 
+class ServerAuth implements GapiAuth {
+  _isSignedIn: boolean;
+  _signInChangedCallback: (isSignedIn: boolean) => void;
+
+  private _loadPromise: Promise<void>;
+
+  /**
+   * Starts the sign-in flow.
+   * If the user has not previously authorized our app, this will redirect to oauth url
+   * to ask the user to select an account and to consent to our use of the scopes.
+   * If the user has previously signed in, the redirect will
+   * happen momentarily before automatically closing.
+   */
+  public async signIn(_doPrompt?: boolean): Promise<void> {
+    const accessToken = await GapiManager.auth.getAccessToken();
+    if (!accessToken) {
+      const currentPath = window.location.pathname;
+      const authLoginUrl = window.location.origin + '/auth/login?page=' + currentPath;
+      window.location.replace(authLoginUrl);
+      return;
+    }
+    this.setIsSignedIn(true);
+  }
+
+  /**
+   * Signs the user out.
+   */
+  public signOut(): Promise<void> {
+    // We sign out by deleting our cookies.
+    Utils.deleteCookie('Datalab-access-token');
+    Utils.deleteCookie('Datalab-access-token-expiry');
+    Utils.deleteCookie('Datalab-refresh-token');
+    Utils.deleteCookie('Datalab-token-type');
+    this.setIsSignedIn(false);
```

Comment:
We should reload or redirect the user to another page here.

Reply:
Maybe we can make that an optional parameter, since I see `signOut` being called for another reason below other than the user initiating an explicit sign out.

Reply:
This method is called from only one place outside this file, from auth-panel, and that location calls reload right after calling signOut, so I don't think we need it here.